src/transformers/pipelines/conversational.py

```diff
@@ -190,23 +190,32 @@ class ConversationalPipeline(Pipeline):
         conversational_pipeline([conversation_1, conversation_2])
     """
 
-    def __init__(self, min_length_for_response=32, minimum_tokens=10, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    min_length_for_response = 32
+    minimum_tokens = 10
+    clean_up_tokenization_spaces = True
 
-        # We need at least an eos_token
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         # assert self.tokenizer.eos_token_id is not None, "ConversationalPipeline tokenizer should have an EOS token set"
         if self.tokenizer.pad_token_id is None:
             self.tokenizer.pad_token = self.tokenizer.eos_token
+        self.max_length = kwargs.get("max_length", self.model.config.max_length)
 
-        self.min_length_for_response = min_length_for_response
-        self.minimum_tokens = minimum_tokens
-
-    def __call__(
-        self,
-        conversations: Union[Conversation, List[Conversation]],
-        clean_up_tokenization_spaces=True,
-        **generate_kwargs
+    def set_parameters(
+        self, min_length_for_response=None, minimum_tokens=None, clean_up_tokenization_spaces=None, **generate_kwargs
     ):
+        if min_length_for_response:
+            self.min_length_for_response = min_length_for_response
+        if minimum_tokens:
+            self.minimum_tokens = minimum_tokens
+        if "max_length" in generate_kwargs:
+            self.max_length = generate_kwargs.get("max_length", self.model.config.max_length)
```

Comment:
This is weird: either always set the attribute using the default:
```py
self.max_length = generate_kwargs.get("max_length", self.model.config.max_length)
```
(with no test) or use the getter if you only want to set the attribute if the result of the test is true:
```py
if "max_length" in generate_kwargs:
    self.max_length = generate_kwargs["max_length"]
```
since it will be in the `generate_kwargs`.